### PR TITLE
Initial implementation of a Bugzilla Patch Crawler.

### PR DESCRIPTION
**Pull request by oSoMoN**

### CODE REVIEW

File: bugzillapatchcrawler.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+
+if __name__ == '__main__':
+    default_instance = 'https://bugzilla.opensuse.org'
+    parser = argparse.ArgumentParser(description='Sample Bugzilla Patch Crawler.')
+    parser.add_argument('-i', metavar='instance',
+                        help=f'an optional Bugzilla instance URL (defaults to {default_instance})')
+    parser.add_argument('-k', metavar='apikey',
+                        help='an optional Bugzilla API key (see https://bugzilla.readthedocs.org/en/5.0/api/core/v1/general.html#authentication)')
+    parser.add_argument('-o', metavar='outfile',
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
    parser.add_argument('--output-file', '-o', metavar='outfile',
```

File: bugzillapatchcrawler.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                        help='an optional file path to write the Patch Store database to (a temporary file will be used if not specified)')
+    parser.add_argument('-t', metavar='delta',
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
    parser.add_argument('--time-delta', '-t', metavar='delta',
```

File: bugzillapatchcrawler.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                        help='an optional time delta (specified in days, defaults to 1)')
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.INFO)
+
+    if args.o is None:
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
    if args.output_file is None:
```

File: bugzillapatchcrawler.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        dbfile, filepath = tempfile.mkstemp(suffix='.db')
+        os.close(dbfile)
+    else:
+        filepath = args.o
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
        filepath = args.ouput_file
```

File: bugzillapatchcrawler.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                    ):
+                        total += 1
+        self.logger.info(f'Done crawling, added {total} patches.')
+        self.timestamp = newts
+
+
+if __name__ == '__main__':
+    default_instance = 'https://bugzilla.opensuse.org'
+    parser = argparse.ArgumentParser(description='Sample Bugzilla Patch Crawler.')
+    parser.add_argument('-i', metavar='instance',
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
    parser.add_argument('--instance', '-i', metavar='instance',
```

File: bugzillapatchcrawler.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                        help=f'an optional Bugzilla instance URL (defaults to {default_instance})')
+    parser.add_argument('-k', metavar='apikey',
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
    parser.add_argument('--api-key', '-k', metavar='apikey',
```

File: bugzillapatchcrawler.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    logging.basicConfig(level=logging.INFO)
+
+    if args.o is None:
+        dbfile, filepath = tempfile.mkstemp(suffix='.db')
+        os.close(dbfile)
+    else:
+        filepath = args.o
+    store = PatchStore(filepath)
+
+    instance = args.i or default_instance
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
    instance = args.instance or default_instance
```

File: bugzillapatchcrawler.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    timestamp = datetime.now() - timedelta(days=int(args.t or 1), seconds=1)
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
    timestamp = datetime.now() - timedelta(days=int(args.time_delta or 1), seconds=1)
```

File: bugzillapatchcrawler.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    crawler = BugzillaPatchCrawler(store, instance, args.k, timestamp)
```
Comment (boiko) [inline bugzillapatchcrawler.py]:
```suggestion
    crawler = BugzillaPatchCrawler(store, instance, args.api_key, timestamp)
```

**Review by boiko — APPROVED** [review]
Code looks pretty good! I just gave some suggestions on using more meaningful names for the arguments (and yet preserving the shortcuts), to keep the code readable, but they are just suggestions still

**Comment by oSoMoN** [comment]
> Code looks pretty good! I just gave some suggestions on using more meaningful names for the arguments (and yet preserving the shortcuts), to keep the code readable, but they are just suggestions still

Thanks for the suggestions, I have applied them.

**Pull request merged**